theatre/studio/src/panels/SequenceEditorPanel/DopeSheet/Right/BasicKeyframedTrack/KeyframeEditor/CurveEditorPopover/CurveEditorPopover.tsx

```diff
@@ -143,315 +124,285 @@ const CurveEditorPopover: React.FC<
-                        1 + svgPadding * 2
-                      }`}
-                      fill="none"
-                      xmlns="http://www.w3.org/2000/svg"
-                    >
-                      <path
-                        d={`M${svgPadding} ${1 + svgPadding} C${
-                          easing[0] + svgPadding
-                        } ${1 - easing[1] + svgPadding} ${
-                          easing[2] + svgPadding
-                        } ${1 - easing[3] + svgPadding} ${
-                          1 + svgPadding
-                        } ${svgPadding}`}
-                        stroke={svgColor}
-                        strokeWidth="0.08"
-                      />
-                      <circle
-                        cx={svgPadding}
-                        cy={1 + svgPadding}
-                        r={svgCircleRadius}
-                        fill={svgColor}
-                      />
-                      <circle
-                        cx={1 + svgPadding}
-                        cy={svgPadding}
-                        r={svgCircleRadius}
-                        fill={svgColor}
-                      />
-                    </svg>
-                  </EasingCurveContainer>
-                  <span>
-                    {useQuery ? (
-                      <span
-                        dangerouslySetInnerHTML={{
-                          __html: presetSearchResults[index].string,
-                        }}
-                      />
-                    ) : (
-                      preset.label
-                    )}
-                  </span>
-                </EasingOption>
+    <Grid>
+      <SearchBox
+        value={filter}
+        placeholder="Search presets..."
+        onPaste={(e) =>
+          // hack to wait for the paste to actually change the e.target.value
+          setTimeout(() => {
+            if (
+              handlesFromCssCubicBezierArgs(
+                (e.target as HTMLInputElement).value,
+              )
+            )
+              editorState.temporarilySetValue(
+                (e.target as HTMLInputElement).value,
               )
-            })}
-          </div>
-        </OptionsContainer>
-      )}
-    </Container>
+          })
+        }
+        onChange={(e) => {
+          setFilter(e.target.value)
+          if (handlesFromCssCubicBezierArgs(e.target.value)) {
+            editorState.temporarilySetValue(e.target.value)
+          }
+        }}
+        ref={inputRef}
+        onKeyDown={onSearchKeyDown}
+      />
+      <OptionsContainer ref={optionsContainerRef}>
+        {displayedPresets.map((preset, index) => (
+          <EasingOption
+            key={preset.label}
+            easing={preset}
+            tabIndex={0}
+            onKeyDown={onEasingOptionKeydown(preset)}
+            ref={optionsRef.current[preset.label]}
+            onMouseOver={() => {
+              isFilterSetBeingUsed.current = true
```

Comment:
Not sure about the UX here. A the user, I'd probably expect the `EasingOption` to only be active while the mouse is hovering it. Once the mouse leaves, it should revert back to whatever was selected before the hover.

https://user-images.githubusercontent.com/593118/166145561-db32dac7-919d-40c8-a3b5-d2522d351f7e.mp4

Reply:
Fixed. Made it so hovering only changes the curve while you are hovering. If you made edits to the curve before hovering, those edits are still present after you stop hovering. This was achieved by adding a second temporary transaction `hoverTransaction` alongside the existing `editTransaction` (previously called `tempTransaction`)

Reply:
See my latest comment on the PR for a video.